snapcraft/plugins/jhbuild.py

```diff
@@ -0,0 +1,383 @@
+
+        self._set_jhbuild_ownership(make_paths)
+
+    def _write_jhbuildrc(self):
+        """
+        _write_jhbuildrc() - write out the jhbuildrc file for build dependency
+        specification
+        """
+        with open(self.jhbuildrc_path, 'w') as jhbuildrc_file:
+            config = '''
+moduleset = {module_set!r}
+modulesets_dir = {modulesets_dir!r}
+tarballdir = {tarballdir!r}
+dvcs_mirror_dir = {dvcs_mirror_dir!r}
+checkoutroot = {checkoutroot!r}
+prefix = {prefix!r}
+buildroot = {buildroot!r}
+use_local_modulesets = True
+skip = [{skip!s}]
+module_autogenargs['gdk-pixbuf'] = '--disable-gio-sniffing'
+extra_prefixes = [{extra_prefixes!s}]
+cflags = {cflags!r}
+'''
+
+            extra_prefixes = [os.path.join(self.project.stage_dir, 'usr'),
+                              os.path.join(self.project.stage_dir, 'usr',
+                                           'local')]
+
+            jhbuildrc_file.write(
+                config.format(module_set=self.options.module_set,
+                              modulesets_dir=self.options.module_set_dir,
+                              tarballdir=self.options.jhbuild_archive or
+                              os.path.join(self.partdir, 'jhbuild',
+                                           'packages'),
+                              dvcs_mirror_dir=os.path.join(
+                                  self.partdir, 'jhbuild', 'mirror'),
+                              checkoutroot=os.path.join(
+                                  self.partdir, 'jhbuild', 'unpacked'),
+                              buildroot=os.path.join(self.builddir, 'jhbuild'),
+                              prefix=os.path.join(self.installdir, 'usr'),
+                              skip=', '.join(
+                                  ['\'%s\'' % module for module in
+                                   self.skip_modules]),
+                              extra_prefixes=', '.join(
+                                  ['\'%s\'' % prefix for prefix in
+                                   extra_prefixes]),
+                              cflags=self.options.cflags,
+                              ))
+
+    def pull(self):
+        self._pull_jhbuild()
+        self._setup_jhbuild()
+
+        self.jhbuild(['sanitycheck'], output=False)
+
+        modules = self.jhbuild(['list'] + self.modules,
+                               output=True).splitlines()
+
+        if 'gtk+' in modules or 'gtk+-3' in modules:
+            self.modules.append('adwaita-icon-theme')
+            self.stage_packages.append('ttf-ubuntu-font-family')
+
+        LOG.info('Pulling modules')
+        self.jhbuild(['update'] + self.modules, output=False)
+
+    def build(self):
+        self._setup_jhbuild()
+
+        LOG.info('Building modules')
+        self.jhbuild(['build'] + self.modules,
+                     cwd=self.builddir, output=False)
+
+        LOG.info('Fixing symbolic links')
+        self.run(['symlinks', '-c', '-d', '-r', '-s', '-v', self.installdir])
```

Comment:
That's not what I meant by my comment-- you may very well need to fix these up. @elopio was just wanting a comment in the code as to why.